src/elm/View/Pin.elm

```diff
@@ -0,0 +1,109 @@
+module View.Pin exposing (..)
+
+import Html exposing (..)
+import Html.Attributes exposing (..)
+import Html.Events exposing (onClick, onInput)
+import I18Next
+import Session.Shared exposing (Shared)
+
+
+type alias Config msg =
+    { labelText : String
+    , inputId : String
+    , inputValue : String
+    , onInputMsg : String -> msg
+    , onToggleMsg : msg
+    , isVisible : Bool
+    , errors : List String
+    }
+
+
+type alias Pin =
+    String
+
+
+isValid : Pin -> Bool
+isValid pin =
+    let
+        hasCorrectLength p =
+            String.length p == 6
+
+        hasOnlyDigits p =
+            String.all Char.isDigit p
```

Comment:
no need to add p here, it can be omitted, currying will make it work!

Reply:
Good point! As for the point-free :)